Upload cards for LFV sample request

jolim97 opened the pull request:
This is the first PR for the signal sample generation for LFV analysis.

Related Analysis : "Search for Lepton Flavour Violation (LFV) in the Top Quark sector with a single muon and tau events at 13 TeV (AN-20-180, not updated)"

sihyunjeon commented:
@jolim97 There are too many cards to go through. Do you have card generating script? If so, can you upload that instead of multiple set of cards?

jolim97 commented:
@sihyunjeon I've uploaded a script for generating cards with default cards (couplings set to 0).

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/LFV_TQMuTau/LFV_ST_TQMuTau/LFV_ST_TQMuTau_customizecards.dat:
```diff
@@ -0,0 +1,2 @@
+set param_card mqt 172.5
```
Is it really called mqt or a typo?

sihyunjeon replied:
https://github.com/cms-sw/genproductions/pull/2825/files#diff-753be9769c41ac94b94a9994e0690d427b6443dbbdd35e2068a37c047c47c2d9R939

This looks like what mass of top quark is named in this model

jolim97 replied:
I can update customizecard with using blocks in parameter cards.
```
set param_card mass 6 172.5
```

agrohsje replied:
No, no. I just wanted to double-check assuming you would know on top of your head and I don't need to verify myself. If that is the name it is the name and all fine.

jolim97 replied:
Okay, I understood. The update version also works, so it would be okay.

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/LFV_TQMuTau/LFV_ST_TQMuTau/LFV_ST_TQMuTau_run_card.dat:
```diff
@@ -0,0 +1,204 @@
+# Number of events and rnd seed                                      *
+# Warning: Do not generate more than 1M events in a single run       *
+#*********************************************************************
+  100 = nevents ! Number of unweighted events requested 
```
I would suggest to increase to 5000 to get reasonable statistics for the pilot run.

jolim97 replied:
I will update the number in the next update.

agrohsje commented:
Why do you need to upload the param_card.dat? Isn't that the one that the model creates by default. What is the flavor scheme? 4fs or 5fs? Did you check that the cuts you are applying have no impact. I know you disabled the decay cuts but some of them are still applied. Did you check?

sihyunjeon commented:
About the param_card, I don't think you have to make customizecards as well. Because, if you look in your param_card in [[link]](https://github.com/cms-sw/genproductions/pull/2825/files#diff-57ad318c1300f5f5a233365997d059fdc237d2546ce483e008836d1d691b870bR939), you can change your mqt there, and also the decay widths yourself. Is there a reason to make customizecards separately?

agrohsje commented:
If the param_card.dat is the default that the model generates, I would really just add customize. CUrrent info indeed is redundant. The customize_card is to keep it simple.

sihyunjeon commented:
@agrohsje that makes sense,
@jolim97 Can you add the parameter setting lines using "set param_card" commands in customize cards and not modifying the parameter card itself and uploading them? You won't need to inject parameter cards for gridpack generation then.

jolim97 commented:
@agrohsje @sihyunjeon Then I will upload only customize card except param card to set parameters in customize cards.
Also, new script for writing customize cards will be updated.

And I have further questions to @agrohsje about following comment.
> Why do you need to upload the param_card.dat? Isn't that the one that the model creates by default. What is the flavor scheme? 4fs or 5fs? Did you check that the cuts you are applying have no impact. I know you disabled the decay cuts but some of them are still applied. Did you check?

1. About param card : I will remove param card and let customizecards to set parameters.
2. About flavour scheme : We are using 4 fs.
3. About cuts : Actually, I don't understand about this question. I found the cut decay product is turned off. What are the things that can I check about decay cuts?

```
   False  = cut_decays    ! Cut decay products 
```

sihyunjeon commented:
False  = cut_decays    ! Cut decay products 

When this is turned on, it will apply cuts on the "decay products" from the onshell resonances. 
 generate p p > t mu+ ta-, (t > b w+, w+ > j j)
So your particles inside the brackets won't get the cuts applied. But the mu+ ta- might get them applied. 

You can check this by computing the cross sections with giving drll = 0.4 and 0.0 with cut_decays=False. I guess with 0.4 (or even higher values) will give smaller cross sections because they will be applied to dR(mu,ta).

https://github.com/cms-sw/genproductions/pull/2825/files#diff-0c80defae607264127d6f6830a97478996622a9bcf36aaeaa907195b6c504c99R123

jolim97 commented:
@sihyunjeon @agrohsje Sorry for late
I have updated the cards with the comments above.
1. Remove param cards -> Set parameters in customizecards (late due to writing a new script).
2. Change the nevents to 5000
3. Check for the decay cuts -> Update with the zero values of drjj, drll, and drjl
* Smaller single top LFV cross section when applying 0.4 drjj, drll, and drjl cut.
* e.g.) 0.07264 +- 0.0002527 pb (with 0.4 cut) -> 0.07277 +- 0.0003128 pb (without cut)
* So the cut values have been changed from 0.4 to 0.0

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/LFV_TQMuTau/LFV_ST_TQMuTau/LFV_ST_TQMuTau_run_card.dat:
```diff
@@ -0,0 +1,204 @@
+#*********************************************************************
+ 20.0  = ptj       ! minimum pt for the jets 
+ 0.0  = ptb       ! minimum pt for the b 
+ 10.0  = ptl       ! minimum pt for the charged leptons 
```
is this and etal also fine? what will be your analysis selection cuts for leptons?

sihyunjeon replied:
I will simply remove all the cuts if there's no specific reasons for it.

jolim97 replied:
They will not affect on our analysis, because we select pt > 30 GeV and |eta| < 2.4 for jets and a muon.
There is no reason for the cuts so I will remove them.

sihyunjeon replied:
@jolim97 Oh nono that's a good enough reason actually. For eta, it will be better to loosen the cut a little bit but ptl is okay.

sihyunjeon commented:
> @sihyunjeon @agrohsje Sorry for late
> I have updated the cards with the comments above.
> 
>     1. Remove param cards -> Set parameters in customizecards (late due to writing a new script).
> 
>     2. Change the nevents to 5000
> 
>     3. Check for the decay cuts -> Update with the zero values of drjj, drll, and drjl
> 
> 
>     * Smaller single top LFV cross section when applying 0.4 drjj, drll, and drjl cut.
> 
>     * e.g.) 0.07264 +- 0.0002527 pb (with 0.4 cut) -> 0.07277 +- 0.0003128 pb (without cut)
> 
>     * So the cut values have been changed from 0.4 to 0.0

Seems like the mu and taus were well separated from the beginning. Though thanks for making the changes.

jolim97 commented:
I have a question on the decay width of tau.
The model has a very small value for the tau decay width (2.25e-12), so I have modified the width to (1.77e-06).
I'm not sure that this is the right value.
As I am understanding, the tau decay process is done in pythia so the decay width value will not be used in madgraph.
Or as like top case, can we set "auto" decay for tau?

agrohsje commented:
auto will allow to calculate the width according to the model used however in your cards the tau will be on its mass shell. you can check its value when interfacing to p8 from the p8 print out.

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/LFV_TQMuTau/LFV_ST_TQMuTau/LFV_ST_TQMuTau_proc_card.dat:
```diff
@@ -0,0 +1,6 @@
+import model /cms/ldap_home/ljw1015/MCProd/mastergenprod/bin/MadGraph5_aMCatNLO/SMEFT_2l2q_UFO_unitary
```
We should add the model to our UFO path and then you can include via 
import model SMEFT_2l2q_UFO_unitary
and then just have the extra model card. Please put it to a public path so Sihyun or myself can copy.

jolim97 replied:
@agrohsje
I've uploaded the model on my cernbox.
https://cernbox.cern.ch/index.php/s/AxylpFmhS3zvoOR

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/LFV_TQMuTau/LFV_TT_TToQMuTau/LFV_TT_TToQMuTau_proc_card.dat:
```diff
@@ -0,0 +1,6 @@
+import model /cms/ldap_home/ljw1015/MCProd/mastergenprod/bin/MadGraph5_aMCatNLO/SMEFT_2l2q_UFO_unitary
```
As above.

jolim97 commented:
I've checked that the model has fake tau decay width (too small), so I set it to auto decay.
With and without auto decay mode, there was a change in cross section.
Also, with uploading the model, I changed the path for the import model and added extra model card.

agrohsje commented:
Thanks @jolim97 . No objections. But you still need the extramodel card to get it from the webspace. I just copied it. Is there.

jolim97 commented:
@agrohsje Yes, I have added extramodel card in the latest commit and checked the gridpack generation works.

agrohsje commented:
Thanks @jolim97 . @sihyunjeon please check and merge. Thanks!

sihyunjeon commented:
merged!

jolim97 commented:
@agrohsje @sihyunjeon I've opened a new PR to fix card writer and proc card for ttbar in https://github.com/cms-sw/genproductions/pull/2846.
